include/courtier/GAsioConsumerT.hpp

```diff
@@ -1191,9 +1193,309 @@ class GAsioConsumerT
+	 //-------------------------------------------------------------------------
+	 // Simplify usage of namespaces
+	 using error_code = boost::system::error_code;
+
+public:
+  GAsioConsumerPT() = default;
+  
+  GAsioConsumerPT(int io_context_pool_size)
+    : m_io_contexts(std::abs(io_context_pool_size))
+    , m_acceptor(m_io_contexts.get())
+    , m_socket(m_io_contexts.get()){
+    std::cout << "-I- GAsioConsumerPT() created with pool size: " <<  io_context_pool_size  << std::endl;
+  }
+  
+  
+  GAsioConsumerPT(const GAsioConsumerPT<processable_type>&) = delete;
+  GAsioConsumerPT(GAsioConsumerPT<processable_type>&&) = delete;
+  GAsioConsumerPT& operator=(const GAsioConsumerPT<processable_type>&) = delete;
+  GAsioConsumerPT& operator=(GAsioConsumerPT<processable_type>&&) = delete;
+  
+  void setNoDelay(){
+    boost::system::error_code set_option_err;
+    boost::asio::ip::tcp::no_delay no_delay(true);
+    m_socket.set_option(no_delay, set_option_err);
+  }
+  
+  void setServerName(const std::string& server) {
+    m_server = server;
+  }
+  
+  std::string getServerName() const {
+    return m_server;
+  }
+  
+  void setPort(unsigned short port) {
+    m_port = port;
+  }
+  
+  unsigned short getPort() const {
+    return m_port;
+  }
+  
+  void setSerializationMode(Gem::Common::serializationMode serializationMode) {
+    m_serializationMode = serializationMode;
+  }
+  
+  Gem::Common::serializationMode getSerializationMode() const {
+    return m_serializationMode;
+  }
+  
+  void setNThreads(std::size_t nThreads) {
+    // Adapt the number of processing threads, if automatic detection was requested
+    if(0 == nThreads) {
+      glogger
+	<< "In GAsioConsumerPT<>::setNThreads(): " << std::endl
+	<< "nThreads was set to 0. m_n_threads will be set to default " << GCONSUMERLISTENERTHREADS << std::endl
+	<< "This replaces the old behaviour where a value of 0 would have" << std::endl
+	<< "resulted in the number of hardware threads being unsed" << std::endl
+	<< GWARNING;
+      
+      m_n_threads = GCONSUMERLISTENERTHREADS;
+    } else {
+      m_n_threads = nThreads;
+    }
+  }
+  
+
+  std::size_t getNProcessingThreads() const {
+    return m_n_threads;
+  }
+  
+  void setMaxReconnects(std::size_t n_max_reconnects) {
+    m_n_max_reconnects = n_max_reconnects;
+  }
+
+  std::size_t getMaxReconnects() const {
+    return m_n_max_reconnects;
+  }
+  
+protected:
+  
+         void shutdown_() override {
+	   // Set the class-wide shutdown-flag
+	   GBaseConsumerT<processable_type>::shutdown_();
+	   
+	   // Make sure context threads may terminate
+	   m_io_contexts.stop();
+
+	 }
+  
+private:
+  
+  void addCLOptions_(
+		     boost::program_options::options_description &visible
+		     , boost::program_options::options_description &hidden
+		     ) override {
+    namespace po = boost::program_options;
+
+    visible.add_options()
+      ("asio_ip", po::value<std::string>(&m_server)->default_value(GCONSUMERDEFAULTSERVER),
```

Comment:
It is not clear to me whether this class is meant as an alternative to the GAsioConsumerT class or is meant to live alongside it? If the latter, than you are duplicating command line options here?